client/keys.go

```diff
@@ -445,7 +445,16 @@ func (hw *httpWatcher) Next(ctx context.Context) (*Response, error) {
 // provided endpoint's path to the root of the keys API
 // (typically "/v2/keys").
 func v2KeysURL(ep url.URL, prefix, key string) *url.URL {
-	ep.Path = path.Join(ep.Path, prefix, key)
+	// This ensures slash at the head if prefix/key is not empty.
```

Comment:
the head is actually in ep.Path. The difference I see here is to ensure we do not remove the trailing slash which path.Join would.